ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift

```diff
@@ -54,6 +54,107 @@ public class KeymanPackage {
     }
   }
 
+  /**
+   * Indicates the distribution level and state of the package.
+   */
+  public enum SupportState: String, Codable {
+    /**
+     * Indicates that the support level for the package is unknown.  Generally occurs when a
+     * package is installed via file-sharing before KeymanEngine is able to perform a package-version check.
+     */
+    case unknown
+
+    /**
+     * Indicates that the package is publicly distributed but no longer maintained.  Indicates that other packages targetting
+     * the same targets are more favored.
+     */
+    case deprecated
+
+    /**
+     * Indicates that this package receives full support and maintenance from the resource development community at large.
+     */
+    case publiclyReleased = "publicly released"
+
+    /**
+     * Indicates that this package is known to not be publicly distributed.
+     */
+    case custom
+  }
+
+  public enum VersionState: String, Codable {
+    /**
+     * KeymanEngine can't track up-to-date versions for custom packages, and we simply don't know before the first update-check for
+     * any models installed from local files.
+     */
+    case unknown
+
+    /**
+     * The installed version of this package matches the latest distributed version.
+     */
+    case upToDate
+
+    /**
+     * There is a publicly-distributed version of the package that is newer than the one currently installed within KeymanEngine.
+     */
+    case needsUpdate
+  }
+
+  public enum InstallationState: String, Codable {
+    /**
+     * While a `KeymanPackage` instance will never return it, some functions that accept a `KeymanPackage.Key` might.
+     * This indicates that the package is neither installed nor being downloaded.
```

Comment:
```suggestion
     * This indicates that the package is neither installed nor being downloaded.
     * A `KeymanPackage` instance will never return this state, some functions that accept a `KeymanPackage.Key` might.
```
(Which functions might?)